<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ComponentDialog</class>
 <widget class="QDialog" name="ComponentDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>819</width>
    <height>499</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>34</x>
     <y>10</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>ID(&amp;I)：</string>
   </property>
   <property name="buddy">
    <cstring>edtID</cstring>
   </property>
  </widget>
  <widget class="QLineEdit" name="edtID">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>10</y>
     <width>291</width>
     <height>21</height>
    </rect>
   </property>
   <property name="maxLength">
    <number>32765</number>
   </property>
  </widget>
  <widget class="QLabel" name="editName_2">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>50</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>名称(&amp;N):</string>
   </property>
   <property name="buddy">
    <cstring>edtName</cstring>
   </property>
  </widget>
  <widget class="QLineEdit" name="edtName">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>50</y>
     <width>291</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QComboBox" name="cbcategory">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>90</y>
     <width>291</width>
     <height>22</height>
    </rect>
   </property>
   <property name="editable">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>90</y>
     <width>53</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>类别(&amp;C):</string>
   </property>
   <property name="buddy">
    <cstring>cbcategory</cstring>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>130</y>
     <width>53</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>设备类型:</string>
   </property>
   <property name="buddy">
    <cstring>cbType</cstring>
   </property>
  </widget>
  <widget class="QComboBox" name="cbType">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>130</y>
     <width>291</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>210</y>
     <width>53</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>设备组:</string>
   </property>
  </widget>
  <widget class="QComboBox" name="cbTeam">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>210</y>
     <width>291</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>170</y>
     <width>53</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>行为(&amp;B):</string>
   </property>
   <property name="buddy">
    <cstring>cbBehavior</cstring>
   </property>
  </widget>
  <widget class="QComboBox" name="cbBehavior">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>170</y>
     <width>291</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>250</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>角色(&amp;R):</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>290</y>
     <width>111</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>置位信号保持时间:</string>
   </property>
   <property name="buddy">
    <cstring>dsbActionKeep</cstring>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="dsbActionKeep">
   <property name="geometry">
    <rect>
     <x>130</x>
     <y>290</y>
     <width>231</width>
     <height>22</height>
    </rect>
   </property>
   <property name="decimals">
    <number>1</number>
   </property>
  </widget>
  <widget class="QLabel" name="labelSubcription">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>60</y>
     <width>351</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>订阅设备(多个设备用&quot;,&quot;分割):</string>
   </property>
   <property name="buddy">
    <cstring>txtSubscription</cstring>
   </property>
  </widget>
  <widget class="QAutoCompletePlainTextEdit" name="txtSubscription">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>80</y>
     <width>421</width>
     <height>151</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>240</y>
     <width>221</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>引用设备(多个设备用&quot;,&quot;分割):</string>
   </property>
   <property name="buddy">
    <cstring>txtReference</cstring>
   </property>
  </widget>
  <widget class="QAutoCompletePlainTextEdit" name="txtReference">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>260</y>
     <width>421</width>
     <height>141</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>410</y>
     <width>181</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>响应的广播(多个广播用&quot;,&quot;分割):</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>10</y>
     <width>53</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>默认值:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>320</y>
     <width>53</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>描述:</string>
   </property>
   <property name="buddy">
    <cstring>txtDescription</cstring>
   </property>
  </widget>
  <widget class="QTextEdit" name="txtDescription">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>340</y>
     <width>341</width>
     <height>151</height>
    </rect>
   </property>
  </widget>
  <widget class="QLineEdit" name="edtRespondBoardcast">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>430</y>
     <width>421</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QLineEdit" name="edtDefaultValue">
   <property name="geometry">
    <rect>
     <x>434</x>
     <y>30</y>
     <width>360</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pbSaveAdd">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>460</y>
     <width>101</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text">
    <string>保存并添加</string>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="pbSaveExit">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>460</y>
     <width>81</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text">
    <string>保存并退出</string>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="pbExit">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>460</y>
     <width>81</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>30</y>
     <width>53</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>{&quot;value&quot;:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_14">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>30</y>
     <width>21</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>}</string>
   </property>
  </widget>
  <widget class="QComboBox" name="cbRole">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>250</y>
     <width>291</width>
     <height>22</height>
    </rect>
   </property>
   <property name="editable">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QAutoCompletePlainTextEdit</class>
   <extends>QPlainTextEdit</extends>
   <header>qautocompleteplaintextedit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>edtID</tabstop>
  <tabstop>edtName</tabstop>
  <tabstop>cbcategory</tabstop>
  <tabstop>cbType</tabstop>
  <tabstop>cbBehavior</tabstop>
  <tabstop>cbTeam</tabstop>
  <tabstop>dsbActionKeep</tabstop>
  <tabstop>txtDescription</tabstop>
  <tabstop>edtDefaultValue</tabstop>
  <tabstop>txtSubscription</tabstop>
  <tabstop>txtReference</tabstop>
  <tabstop>edtRespondBoardcast</tabstop>
  <tabstop>pbSaveAdd</tabstop>
  <tabstop>pbSaveExit</tabstop>
  <tabstop>pbExit</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
